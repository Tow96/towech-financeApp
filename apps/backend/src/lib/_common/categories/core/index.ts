export * from './category-aggregate';
export * from './category-events';
export * from './i-category-repository';
export * from './subcategory-entity';
export * from './subcategory-events';
