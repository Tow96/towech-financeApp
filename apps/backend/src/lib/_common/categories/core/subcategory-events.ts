// Slice packages
import { CategoryType } from './category-aggregate';

// Internal references
import { DomainEvent, DomainEventProps } from '../../primitives';

// ----------------------------------------------
export class SubCategoryCreatedEvent extends DomainEvent {
  readonly userId: string;
  readonly categoryId: string;
  readonly name: string;
  readonly type: CategoryType;

  constructor(props: DomainEventProps<SubCategoryCreatedEvent>) {
    super(props);
    this.userId = props.userId;
    this.categoryId = props.categoryId;
    this.name = props.name;
    this.type = props.type;
  }
}

export class SubCategoryUpdatedEvent extends DomainEvent {
  readonly categoryId: string;
  readonly type: CategoryType;
  readonly name: string;

  constructor(props: DomainEventProps<SubCategoryUpdatedEvent>) {
    super(props);
    this.categoryId = props.categoryId;
    this.type = props.type;
    this.name = props.name;
  }
}

export class SubCategoryRemovedEvent extends DomainEvent {
  readonly categoryId: string;

  constructor(props: DomainEventProps<SubCategoryRemovedEvent>) {
    super(props);
    this.categoryId = props.categoryId;
  }
}
