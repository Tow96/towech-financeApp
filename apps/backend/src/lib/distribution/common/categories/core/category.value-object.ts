// App packages
import { ValueObject } from '../../../../_common/primitives/value-object.base';

export enum CategoryType {
  income = 'INCOME',
  expense = 'EXPENSE',
  transfer = 'TRANSFER',
}

interface CategoryProps {
  _id: string;
  _userId: string;
  _type: CategoryType;
}

export class Category extends ValueObject<CategoryProps> {
  get id() {
    return this.props._id;
  }

  get userId() {
    return this.props._userId;
  }

  get type() {
    return this.props._type;
  }
}
