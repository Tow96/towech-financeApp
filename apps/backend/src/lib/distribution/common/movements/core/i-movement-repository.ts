import { MovementAggregate } from './movement-aggregate';

export abstract class IMovementRepository {
  abstract getMovementOwner(id: string): Promise<string | null>;
  abstract getById(id: string): Promise<MovementAggregate | null>;

  abstract saveChanges(aggregate: MovementAggregate): Promise<void>;
}
