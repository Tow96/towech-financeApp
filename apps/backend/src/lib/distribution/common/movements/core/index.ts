export * from './category.value-object';
export * from './i-movement-repository';
export * from './movement-aggregate';
export * from './summary-item.value-object';
export * as events from './movement-events';
