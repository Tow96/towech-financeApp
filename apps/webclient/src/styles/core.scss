// The theming functions are heavily taken from angular material, but in a lighter way
// Original code: https://material.angular.io/
@use 'sass:map';

@function get-palette-hue($palette, $hue) {
  @return map.get($palette, $hue);
}

@function get-palette-hue-contrast($palette, $hue) {
  @return map.get(map.get($palette, contrast), $hue);
}
