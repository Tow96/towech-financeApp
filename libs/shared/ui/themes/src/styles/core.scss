// The theming functions are heavily taken from angular material, but in a lighter way
// Original code: https://material.angular.io/
@use 'sass:map';

@function get-palette-hue($palette, $hue) {
  @return map.get($palette, $hue);
}

@function get-palette-hue-contrast($palette, $hue) {
  @return map.get(map.get($palette, contrast), $hue);
}

@function define-theme(
  $background,
  $text,
  $primary,
  $accent,
  $success: null,
  $warning: null,
  $error: null
) {
  @if $background == null or $text == null {
    @error 'Background and text colors need to be defined';
  }

  @if $primary == null or $accent == null {
    @error 'primary and accent colors need to be defined palettes';
  }

  @if not map.get($primary, contrast) {
    @error 'Primary is not a palette';
  }

  @if not map.get($accent, contrast) {
    @error 'Accent is not a palette';
  }

  @if $success != null and not map.get($success, contrast) {
    @error 'Success is not a palette';
  }

  @if $error != null and not map.get($error, contrast) {
    @error 'Error is not a palette';
  }

  @if $warning != null and not map.get($warning, contrast) {
    @error 'Warning is not a palette';
  }

  $result: (
    color: (
      text: $text,
      background: $background,
      primary: $primary,
      accent: $accent,
      success: if($success != null, $success, $accent),
      warning: if($warning != null, $warning, $accent),
      error: if($error != null, $error, $accent),
    ),
  );

  @return $result;
}

@mixin color($theme) {
  $colors: map.get($theme, color);
  $primary: map.get($colors, primary);

  :root {
    background-color: map.get($colors, background);
    color: map.get($colors, text);
  }

  div.container {
    background-color: get-palette-hue($primary, 500);
  }
}
