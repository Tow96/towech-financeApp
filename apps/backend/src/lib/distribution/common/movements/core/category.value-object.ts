// App packages
import { ValueObject } from '../../../../_common/primitives';
import { CategoryType } from '../../../../_common/categories';

interface CategoryProps {
  _id: string;
  _userId: string;
  _type: CategoryType;
}

export class Category extends ValueObject<CategoryProps> {
  get id() {
    return this.props._id;
  }

  get userId() {
    return this.props._userId;
  }

  get type() {
    return this.props._type;
  }

  protected validate(props: CategoryProps): void {
    if (props._id === '' || props._userId === '') throw new Error('Empty values');
  }
}
