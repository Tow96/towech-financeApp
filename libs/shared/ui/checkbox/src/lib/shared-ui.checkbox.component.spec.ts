// Libraries
import { ComponentFixture, TestBed } from '@angular/core/testing';
// Tested elements
import { SharedCheckboxComponent } from './shared-ui-checkbox.component';
import { NGRX_FORM_VIEW_ADAPTER } from 'ngrx-forms';

describe('Shared Input Component', () => {
  let component: SharedCheckboxComponent;
  let fixture: ComponentFixture<SharedCheckboxComponent>;
  let compiled: HTMLElement;

  beforeEach(() => {
    fixture = TestBed.createComponent(SharedCheckboxComponent);
    component = fixture.componentInstance;
    // component.type = 'text';
    component.label = 'TEST';

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('Should be defined', () => expect(component).toBeTruthy());

  it('Must match the snapshot', () => expect(compiled).toMatchSnapshot());

  it('Should implement the NGRX_FORM_VIEW_ADAPTER', () => {
    expect(fixture.debugElement.injector.get(NGRX_FORM_VIEW_ADAPTER)).toBeTruthy();
  });

  it('The changed and touched functions must do nothing at start', () => {
    expect(component.changed(false)).toBeUndefined();
    expect(component.touched()).toBeUndefined();
  });

  describe('When setViewValue is called', () => {
    it('Should update the value', () => {
      component.setViewValue(true);
      expect(component.value).toBe(true);

      component.setViewValue(false);
      expect(component.value).toBe(false);
    });
  });

  describe('When setOnChangeCallback is called', () => {
    it('Should update the changed function', () => {
      const newChanged = (value: boolean) => {
        console.log(value);
      };

      component.setOnChangeCallback(newChanged);
      expect(component.changed).toBe(newChanged);
    });
  });

  describe('When setOnTouchedCallback is called', () => {
    it('Should update the touched function', () => {
      const newTouched = () => {
        console.log('NEW FUNCTION');
      };

      component.setOnTouchedCallback(newTouched);
      expect(component.touched).toBe(newTouched);
    });
  });

  describe('When setIsDisabled is called', () => {
    it('Should set the disabled value', () => {
      component.setIsDisabled(true);
      expect(component.disabled).toBe(true);
      component.setIsDisabled(false);
      expect(component.disabled).toBe(false);
    });
  });

  describe('When on change is called', () => {
    it('Should call the changed function with the new value', () => {
      const event = { target: { checked: true } } as any;

      const spy = jest.spyOn(component, 'changed');
      component.onChange(event);
      expect(spy).toHaveBeenCalledWith(true);

      jest.clearAllMocks();
      event.target.checked = false;
      component.onChange(event);
      expect(spy).toHaveBeenCalledWith(false);
    });
  });
});
