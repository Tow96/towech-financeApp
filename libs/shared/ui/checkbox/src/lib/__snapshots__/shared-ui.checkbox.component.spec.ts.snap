// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Input Component Must match the snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="checkbox"
  >
    <input
      id="towech-checkbox"
      type="checkbox"
    />
    <label
      for="towech-checkbox"
    >
      TEST
    </label>
  </div>
</div>
`;
